### PR TITLE
Fix Incorrect updated affine in `NrrdReader` and update docstring in `ITKReader`

### LINKED ISSUE #7371: Recommended reader for `.nrrd` files? `NRRDReader` gives wrong output for non-diagonal orientations
**Describe the bug**

The default reader for '.nrrd' file extensions is the `NRRDReader` as shown in the docs, (https://docs.monai.io/en/stable/transforms.html#loadimage) as well as used by `LoadImage` when trying to load the file type (all dependencies are installed) 

However, `NRRDReader` does not work accurately for non-diagonal orientation matrices. 

`ITKReader` seems to work reliably for each use case, whereas `NRRDReader` doesn't. 

Is there a reason why `NRRDReader` is preferred over the `ITKReader`? Also, since `ITKReader` reads nifti files, why is it not the default reader selected for all such formats?

**To Reproduce**

Code using the default `NRRDReader`:
```python
import monai
import matplotlib.pyplot as plt

nrrd_datalist = ["path_to_nrrd_file"]

transform = monai.transforms.Compose([
                                      monai.transforms.LoadImage(image_only=True, ensure_channel_first=True),
                                      monai.transforms.DataStats(additional_info=lambda x: x.affine, prefix="On load"),
                                      monai.transforms.Orientation(axcodes="LPS"), 
                                      monai.transforms.Transpose(indices=[0, 3, 2, 1]),
                                      monai.transforms.DataStats(additional_info=lambda x: x.affine, prefix="After transform")
                                      ])

out = transform(nrrd_datalist)


# Get the mid indexes for each dimension
img = out[0][0]
mid_dim0 = img.shape[0] // 2
mid_dim1 = img.shape[1] // 2
mid_dim2 = img.shape[2] // 2

# Plot mid-axial, sagittal and coronal views
fig, axs = plt.subplots(1, 3, figsize=(15, 5))
axs[0].imshow(img[mid_dim0, :, :], cmap='gray')
axs[0].set_title('Mid dim0')
axs[1].imshow(img[:, mid_dim1, :], cmap='gray')
axs[1].set_title('Mid dim1')
axs[2].imshow(img[:, :, mid_dim2], cmap='gray')
axs[2].set_title('Mid dim2')
plt.show()
```
On load statistics:
Type: <class 'monai.data.meta_tensor.MetaTensor'> torch.float32
Shape: torch.Size([1, 512, 512, 78])
Value range: (-1024.0, 3071.0)
Additional info: tensor([[   0.0000,   -0.4941,    0.0000, -262.2529],
        [   0.4941,    0.0000,    0.0000,   56.2529],
        [   0.0000,    0.0000,    1.5000,   59.5000],
        [   0.0000,    0.0000,    0.0000,    1.0000]], dtype=torch.float64)

After transform statistics:
Type: <class 'monai.data.meta_tensor.MetaTensor'> torch.float32
Shape: torch.Size([1, 78, 512, 512])
Value range: (-1024.0, 3071.0)
Additional info: tensor([[  -0.4941,    0.0000,    0.0000, -262.2529],
        [   0.0000,   -0.4941,    0.0000,  308.7588],
        [   0.0000,    0.0000,    1.5000,   59.5000],
        [   0.0000,    0.0000,    0.0000,    1.0000]], dtype=torch.float64)

image

Code using `ITKReader`:
```python
import monai
import matplotlib.pyplot as plt

nrrd_datalist = ["path_to_nrrd_file"]

transform = monai.transforms.Compose([
                                      monai.transforms.LoadImage(image_only=True, ensure_channel_first=True, reader="ITKReader"),
                                      monai.transforms.DataStats(additional_info=lambda x: x.affine, prefix="On load"),
                                      monai.transforms.Orientation(axcodes="LPS"), 
                                      monai.transforms.Transpose(indices=[0, 3, 2, 1]),
                                      monai.transforms.DataStats(additional_info=lambda x: x.affine, prefix="After transform")
                                      ])

out = transform(nrrd_datalist)


# Get the mid indexes for each dimension
img = out[0][0]
mid_dim0 = img.shape[0] // 2
mid_dim1 = img.shape[1] // 2
mid_dim2 = img.shape[2] // 2

# Plot mid-axial, sagittal and coronal views
fig, axs = plt.subplots(1, 3, figsize=(15, 5))
axs[0].imshow(img[mid_dim0, :, :], cmap='gray')
axs[0].set_title('Mid dim0')
axs[1].imshow(img[:, mid_dim1, :], cmap='gray')
axs[1].set_title('Mid dim1')
axs[2].imshow(img[:, :, mid_dim2], cmap='gray')
axs[2].set_title('Mid dim2')
plt.show()

```
On load statistics:
Type: <class 'monai.data.meta_tensor.MetaTensor'> torch.float32
Shape: torch.Size([1, 512, 512, 78])
Value range: (-1024.0, 3071.0)
Additional info: tensor([[   0.0000,    0.4941,    0.0000, -262.2529],
        [  -0.4941,    0.0000,    0.0000,   56.2529],
        [   0.0000,    0.0000,    1.5000,   59.5000],
        [   0.0000,    0.0000,    0.0000,    1.0000]], dtype=torch.float64)

After transform statistics:
Type: <class 'monai.data.meta_tensor.MetaTensor'> torch.float32
Shape: torch.Size([1, 78, 512, 512])
Value range: (-1024.0, 3071.0)
Additional info: tensor([[-0.4941,  0.0000,  0.0000, -9.7471],
        [ 0.0000, -0.4941,  0.0000, 56.2529],
        [ 0.0000,  0.0000,  1.5000, 59.5000],
        [ 0.0000,  0.0000,  0.0000,  1.0000]], dtype=torch.float64)

image

**Environment**
[monai_env_config.txt](https://github.com/Project-MONAI/MONAI/files/13846385/monai_env_config.txt)

### LINKED ISSUE #7414: Incorrect updated affine in `NrrdReader`
- The docstring for `reverse_indexing` in `ITKReader` is wrong. 
https://github.com/Project-MONAI/MONAI/blob/abe2e3151e77fdbcad68268111aaabf1e73c3dbe/monai/data/image_reader.py#L170-L173
https://github.com/Project-MONAI/MONAI/blob/abe2e3151e77fdbcad68268111aaabf1e73c3dbe/monai/data/image_reader.py#L368-L372
https://github.com/InsightSoftwareConsortium/ITK/blob/f5aab1f68e53061264f4f6f3af5e0d49c5f7481d/Wrapping/Generators/Python/itk/support/extras.py#L351-L354

- In `NrrdReader`, even the header have been updated after `_convert_f_to_c_order`, it dosen't pass to the `_get_affine`, so the affine is still not updated.
https://github.com/Project-MONAI/MONAI/blob/abe2e3151e77fdbcad68268111aaabf1e73c3dbe/monai/data/image_reader.py#L1324-L1326

### PR DESCRIPTION
**Pull request by KumoLiu**
Fixes #7414
Fixes #7371

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/data/image_reader.py [hunk 1]
```diff
@@ -1323,7 +1323,7 @@ def get_data(self, img: NrrdImage | list[NrrdImage]) -> tuple[np.ndarray, dict]:
             header = dict(i.header)
             if self.index_order == "C":
                 header = self._convert_f_to_c_order(header)
-            header[MetaKeys.ORIGINAL_AFFINE] = self._get_affine(i)
+            header[MetaKeys.ORIGINAL_AFFINE] = self._get_affine(header)
```
Comment (ericspod) [inline monai/data/image_reader.py:1326]: I'm guessing the fix here is that the header being used has been passed through `self._convert_f_to_c_order` whereas this wasn't used in the original?

**Review by ericspod — APPROVED** [review]
I think this fixes the issue and documents things properly. @maunzzz @surajpaib does this solve the issue for you?

File: monai/data/image_reader.py (hunk 1 above)
Comment (KumoLiu) [reply monai/data/image_reader.py:1326]: Yes, and also affine is a copy of the original.
https://github.com/Project-MONAI/MONAI/pull/7415/files#diff-f9fa1fca729a2ee3040291d8ba9734458f2f8af4ef9b00fe7108ded2ec6006faL1333

**Pull request merged**